vm/src/type.h

```diff
@@ -0,0 +1,56 @@
+#ifndef BORGES_TYPE_H
+#define BORGES_TYPE_H
+
+#include <cstddef>
+#include <type_traits>
+#include <vector>
+
+namespace Borges {
+
+class Serializer;
+class Deserializer;
+
+// Identifies the run-time type of a value.
+enum class Type : std::size_t {
+    boolean,
+    number,
+    vector_2,
+    vector_3,
+    matrix_3,
+    matrix_4,
+    boolean_array,
+    number_array,
+    vector_2_array,
+    vector_3_array,
+    matrix_3_array,
+    matrix_4_array
+};
+
```

Comment:
we don't need support for (de)serialization of the vec2 and matrix3 (and _array) classes imo.